tests/cbmc/proofs/s2n_is_base64_char/s2n_is_base64_char_harness.c

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "api/s2n.h"
+
+#include "stuffer/s2n_stuffer.h"
+
+#include <assert.h>
+
+void s2n_is_base64_char_harness() {
+    char c;
+    int result = s2n_is_base64_char(c);
+    assert(result == 0 || result == 1);
```

Comment:
What exactly are we proving here? I read this as:

> `s2n_is_base64_char` takes any `char` value and returns `true` or `false`.

But, that's already proven to be true by the type system.

(speaking of, can you change the return value to `bool`, rather than `int`?)

Reply:
@camshaft This code is going to check the following properties in the implementation of `s2n_is_base64_char` function:
```Makefile
CHECKFLAGS += --bounds-check
CHECKFLAGS += --conversion-check
CHECKFLAGS += --div-by-zero-check
CHECKFLAGS += --float-overflow-check
CHECKFLAGS += --nan-check
CHECKFLAGS += --pointer-check
CHECKFLAGS += --pointer-overflow-check
CHECKFLAGS += --signed-overflow-check
CHECKFLAGS += --undefined-shift-check
CHECKFLAGS += --unsigned-overflow-check
CHECKFLAGS += --unwinding-assertions
```
This harness takes a non-deterministic `char` value, injects it into the function, and asserts the result, so we can prove that the implementation holds the above properties for any `char` value. The assertion seems a bit unnecessary, specially if we change the return type to `bool` (I could remove it or assert another functional property). In this particular function, it's interesting to check arithmetic conversion errors.